### PR TITLE
Make GPU scans process multiple items per thread

### PR DESCRIPTION
**Pull request by shreyas-omkar**
GPU scans currently process two items per thread. This PR makes that configurable with an `items_per_thread` keyword and uses eight by default where the element type fits within the 32 KiB shared-memory baseline. The value does not need to be a power of two.

Each thread scans a blocked run after a coalesced load, then the block scans the per-thread totals. On an RTX 5080 with 64M `Int32` elements, this reduced `ScanPrefixes` from 2.00 ms to 1.31 ms and `DecoupledLookback` from 10.4 ms to 3.8 ms.

This also fixes multi-block exclusive `DecoupledLookback` scans. After inclusive carry propagation, two epilogue kernels save one boundary value per tile and shift each tile in place; no full-array temporary is allocated. `ScanPrefixes` remains the default.

Validated with the CUDA and OpenCL accumulate test suites.

### CODE REVIEW

**Comment by shreyas-omkar** [comment]
### Cross-backend, Float32
| Backend (GPU) | n | ipt=2 (GB/s)| ipt=4 (GB/s) | ipt=8 (GB/s) |
|---|---:|---:|---:|---:|
| **CUDA** RTX 5080 | 1M | 70.5 | 81.0 | 87.5 |
| | 16M | 489 | 578 | 578 |
| | 64M | 269 | 413 | 408 |
| **AMDGPU** RX 9060 XT | 1M | 97.5 | 106.7 | 90.2 | 
| | 16M | 139 | 143 | 137 | 
| | 64M | 57.0 | 146 | 137 |
| | 128M | 29.8 | 148 | 139 |
| **Metal** Apple M5 | 1M | 10.4 | 10.1 | 10.5 |
| | 16M | 12.4 | 55.5 | 60.2 | 
| | 64M | 6.3 | 20.7 | 40.8 | 
| **POCL** Ryzen 7700X* | 1M | 3.1 | 5.2 | 7.8 | 
| | 8M | 3.3 | 6.7 | 9.4 |

File: test/generic/accumulate.jl [hunk 1]
```diff
@@ -33,22 +33,26 @@ TEST_DL && push!(ALGS, AK.DecoupledLookback())
...
             @test Array(y) == (cumsum(xh) .- xh) .+ init
         end
     end
 
+    # The default limits shared-memory use for wide element types.
+    xh = ComplexF64.(1:4097)
+    y = array_from_host(xh)
+    AK.accumulate!(+, y; prefer_threads, init=0.0 + 0.0im, alg)
+    @test Array(y) == cumsum(xh)
+
```
Comment (christiangnrd) [inline test/generic/accumulate.jl:57]: This should use a different eltype or avoid testing on unsupported backends

Comment (christiangnrd) [reply test/generic/accumulate.jl:57]: Seems like this got fixed after I started typing but before I submitted

Comment (maleadt) [reply test/generic/accumulate.jl:57]: Yeah I noticed the oneAPI.jl CI failure.
